docs/spec/annotations.rst

```diff
@@ -353,10 +349,14 @@ types cannot be specified::
 Annotating instance and class methods
 -------------------------------------
 
-In most cases the first argument of class and instance methods
-does not need to be annotated, and it is assumed to have the
-type of the containing class for instance methods, and a type object
-type corresponding to the containing class object for class methods.
+In most cases the first argument of instance and class methods
+(conventionally named ``self`` or ``cls``) does not need to be annotated.
+
+If the argument is not annotated, then for instance methods it is
+assumed to have the type of the containing class or :ref:`Self
+<self>`, and for class methods the type object type corresponding to
+the containing class object or ``type[Self]``.
```

Comment:
I don't think it is accurate to make an explicit equivalence here between the type `Self` and "the type of the containing class". This is not actually what the specification for `Self` says: `Self` is not "the type of the containing class," it is "a typevar with a bound of the containing class". This distinction is important for how the `Self` type is handled when used in other arguments, and in inheritance cases. 

I suggest we simply don't attempt a mini-definition of the meaning of `Self` here, and allow the `Self` spec to handle this in depth.

```suggestion
assumed to have the type :ref:`Self <self>`, and for class methods ``type[Self]``.
```

Reply:
Oh! I see how that could be the interpretation of the "or" here. If that's the intention, then I suggest we make it a bit more explicit. (I'll make a separate suggested edit, since it would involve one more line than I included in this one.)

Reply:
Personally I think inferring `Self` is the better option here, and we could also consider just specifying that, but I see that may be a bigger change than was intended in this PR.